test/signal.spec.ts

```diff
@@ -28,6 +28,25 @@ test('Signal listeners should received dispatched payloads', t => {
     t.end();
 });
 
+test('All listeners should receive dispatched payloads even with exceptions', t => {
+    const signal = new Signal<string>();
+
+    const sentPayloads = ['a', 'b', 'c'];
+
+    const receivedPayloadsListener2: string[] = [];
+
+    signal.add(() => {
+        throw new Error('');
+    });
+
+    signal.add((payload: string) => receivedPayloadsListener2.push(payload));
+    sentPayloads.forEach(payload => signal.dispatch(payload));
+
+    t.deepEqual(receivedPayloadsListener2, sentPayloads);
```

Comment:
I am thinking in this case we do not know in which order they dispatch, so it might not test as thoroughly as intended. There is a chance that the listener that throws is dispatched second.

Maybe a way to fix this would be to add two listeners, each one adds to a received payloads array and then throws, then for sure you could see that one was always dispatched after an exception.

Reply:
I agree. Will enhance the test to take that into account.